src/main/kotlin/com/atlassian/performance/tools/report/api/judge/Verdict.kt

```diff
@@ -14,20 +14,42 @@ class Verdict internal constructor(
     val reports: List<JUnitReport>,
     val failedActions: List<ActionType<*>>
 ) {
+    private companion object {
+        private val logger = LogManager.getLogger(this::class.java)
+    }
 
+    @Deprecated("Use Builder")
     constructor(
         reports: List<JUnitReport>
     ) : this(
         reports = reports,
         failedActions = emptyList()
     )
 
-    private val logger = LogManager.getLogger(this::class.java)
+    /**
+     * Since 3.12.0
+     */
+    class Builder(private val reports: List<JUnitReport>) {
+        private var failedActions: List<ActionType<*>> = emptyList()
+
+        fun failedActions(failedActions: List<ActionType<*>>): Builder {
+            this.failedActions = failedActions
+            return this
+        }
+
+        fun build(): Verdict {
+            return Verdict(
+                reports = reports,
```

Comment:
Mutator/`with`er for `reports` is missing. PS. Worth considering an explicit `addReport` if you want it to be append-only.